src/utils/algos.js

```diff
@@ -0,0 +1,17 @@
+export function equalsArray(a, b) {
+    if (!b || !a) return false;
+    if (b === a) return true;
+    if (a.length !== b.length) return false;
+
+    for (var i = 0, l = a.length; i < l; i++) {
+        // Check if we have nested arrays
+        if (a[i] instanceof Array && b[i] instanceof Array) {
+            // recurse into the nested arrays
+            if (!a[i].equals(b[i])) return false;
```

Comment:
I did it first, but noticed it would not be really needed.
So I simplified, while planning to let it be configurable via an optional boolean or an optional max_depth.
But forget to do so. Which option would you go ?

Reply:
I think you can let it like that and just edit the comment

Reply:
Done